File: frontend/src/utils/currency.js
export const CURRENCIES = [
  { code: 'MAD', symbol: 'MAD', label: 'Dirham (MAD)' },
  { code: 'EUR', symbol: '€', label: 'Euro (€)' },
  { code: 'USD', symbol: '$', label: 'Dollar ($)' },
];

// Profiles saved before the currency field existed were analysed in euros.
const LEGACY_CURRENCY = 'EUR';

// Revenue buckets per currency. Same index = same order of magnitude, so switching
// currency maps the selected bucket to its equivalent instead of losing it.
export const REVENUE_RANGES = {
  EUR: ['<500k', '500k-2M', '2-10M', '10-50M', '50-250M', '>250M'],
  USD: ['<500k', '500k-2M', '2-10M', '10-50M', '50-250M', '>250M'],
  MAD: ['<5M', '5-20M', '20-100M', '100-500M', '500M-2,5Md', '>2,5Md'],
};

export const currencyOf = (profile) => profile?.currency || LEGACY_CURRENCY;

export const currencySymbol = (code) =>
  CURRENCIES.find((c) => c.code === code)?.symbol || code;

export function mapRevenueRange(range, from, to) {
  const i = (REVENUE_RANGES[from] || []).indexOf(range);
  return i >= 0 ? REVENUE_RANGES[to][i] : null;
}

File: ai-python/app/currency.py
"""
Project currency (chosen in onboarding, profile["currency"]).

Every amount the agents write must use this currency. Profiles saved before the field
existed have no currency: they were analysed in euros, so they fall back to EUR.
"""
from __future__ import annotations
from typing import Optional

_CURRENCIES = {
    # code: (French name, English name, "millions" unit written in the outputs)
    "MAD": ("dirham marocain", "Moroccan dirham", "M MAD"),
    "EUR": ("euro", "euro", "M€"),
    "USD": ("dollar américain", "US dollar", "M$"),
}
LEGACY_CURRENCY = "EUR"


def currency_code(profile: Optional[dict]) -> str:
    code = str((profile or {}).get("currency") or "").strip().upper()
    return code if code in _CURRENCIES else LEGACY_CURRENCY


def currency_label(profile: Optional[dict], is_en: bool = False) -> str:
    code = currency_code(profile)
    return f"{code} ({_CURRENCIES[code][1 if is_en else 0]})"


def million_unit(profile: Optional[dict]) -> str:
    return _CURRENCIES[currency_code(profile)][2]
